### PR TITLE
transactions commands and send with a custom tag

### PR DESCRIPTION
**Pull request by valerio-vaccaro**
Some update and improvements:
- submodules are now all fetched via HTTP,
- I solved an issue on tag field and connected to the transaction function,
- I added  a transactions command useful in order to search transaction connected to an address.

### CODE REVIEW

File: main/wallet_system.c [hunk 1]
```diff
@@ -388,9 +395,18 @@ static int fn_send(int argc, char **argv) {
     security = 2;
   }
 
+  char padded_tag[28];
```
Comment (oopsmonk) [inline main/wallet_system.c]:
```suggestion
  char padded_tag[NUM_TRYTES_TAG+1];
```

File: main/wallet_system.c [hunk 2]
```diff
@@ -388,9 +395,18 @@ static int fn_send(int argc, char **argv) {
...
+  size_t tag_size = strlen(tag);
+  convertToUpperCase((char *)tag, tag_size);
+  for (size_t i = 0; i < 27; i++){
```
Comment (oopsmonk) [inline main/wallet_system.c]:
```suggestion
  for (size_t i = 0; i < NUM_TRYTES_TAG; i++){
```

File: main/wallet_system.c [hunk 3]
```diff
@@ -388,9 +395,18 @@ static int fn_send(int argc, char **argv) {
...
+    if (i < tag_size) padded_tag[i] = tag[i];
+    else padded_tag[i] = '9';
+  }
+  padded_tag[27] = '\0';
```
Comment (oopsmonk) [inline main/wallet_system.c]:
```suggestion
  padded_tag[NUM_TRYTES_TAG] = '\0';
```

File: main/wallet_system.c [hunk 4]
```diff
@@ -467,6 +484,86 @@ static void register_send() {
...
+
+  int nerrors = arg_parse(argc, argv, (void **)&get_transactions_args);
+  if (nerrors != 0) {
+    arg_print_errors(stderr, get_transactions_args.end, argv[0]);
+    return 1;
+  }
+
+  char const *address_ptr = get_transactions_args.address->sval[0];
+
+#if 0
```
Comment (oopsmonk) [inline main/wallet_system.c]: remove

**Review by oopsmonk — CHANGES_REQUESTED** [review]
LGTM, it needs some slight changes, thanks

File: main/wallet_system.c (hunk 4 above)
Comment (valerio-vaccaro) [reply main/wallet_system.c]: address pointer is requested by the following code, what  I need to remove?

Comment (oopsmonk) [reply main/wallet_system.c]: I meant the unused code in `#if 0`

Comment (valerio-vaccaro) [reply main/wallet_system.c]: done

**Review by oopsmonk — APPROVED** [review]

**Comment by oopsmonk** [comment]
many thanks :-)

**Pull request merged**